### DISCUSSION TITLE
Stablecoin Project: ERC20Mock problem (Probably cannot find the correct Mock contract)

### DISCUSSION THREAD

**Discussion by boringslav** [Cyfrin/foundry-full-course-cu#42] [Q&A] [answered]
When creating the mocks for WETH and WBTC we are using the following code:
```solidity
  // Contract import statement
  import {ERC20Mock} from "@openzeppelin/contracts/mocks/ERC20Mock.sol";
  ...
  // Code 1:1 as the code that Patrick wrote to create mocks
  ERC20Mock wethMock = new ERC20Mock("WETH", "WETH", msg.sender, 1000e8);
  ERC20Mock wbtcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);
```
In this code, we are providing name, symbol, initialAccount and initialBalance (function arguments).
The mock contract that I found [link here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/mocks/token/ERC20Mock.sol) doesn't have any parameters for the constructor. I think I am importing the wrong contract but I cannot manage to find the one with the desired parameters in the constructor.
```solidity
// The Mock contract that I found
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

import {ERC20} from "../token/ERC20/ERC20.sol";

contract ERC20Mock is ERC20 {
    constructor() ERC20("ERC20Mock", "E20M") {}

    function mint(address account, uint256 amount) external {
        _mint(account, amount);
    }

    function burn(address account, uint256 amount) external {
        _burn(account, amount);
    }
}
```

**Comment by 0xCiphky** [comment, marked as answer]
The issue might be from using the latest dependency that has updated the mock contract. To solve this, remove the openzeppelin dependency and install the specific version that is being used.

forge remove lib/openzeppelin-contracts
forge install openzeppelin/openzeppelin-contracts@V4.8.3 --no-commit

There's probably a easier way to do this.

**Reply by cromewar** [reply]
Does this makes sense for you @boringslav? As @areezladhani mentioned the Mock version on 4.8.3 is slightly different as you can see [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.3/contracts/mocks/ERC20Mock.sol).

**Reply by boringslav** [reply]
It fixed it, thank you @areezladhani @cromewar ! Also for the people that will get the same issue `forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit` the v for the version is not capital.
